activerecord/test/cases/finder_test.rb

```diff
@@ -202,10 +202,15 @@ def test_exists_with_nil_arg
     assert_equal true, Topic.first.replies.exists?
   end
 
-  # Ensure +exists?+ runs without an error by excluding distinct value.
+  # ensures +exists?+ runs without an error by excluding distinct value
   # See https://github.com/rails/rails/pull/26981.
-  def test_exists_with_order_and_distinct
-    assert_equal true, Topic.order(:id).distinct.exists?
+  def test_exists_with_distinct
+    assert_equal true, Topic.distinct.exists?
+  end
+
+  # ensures +exists?+ runs without an error by excluding order value
+  def test_exists_with_order
+    assert_equal true, Topic.order("invalid sql here").exists?
```

Comment:
I still wonder why you want to run `Model.outstanding.joins(:another_model_with_created_at).exists?` because `Model.outstanding.joins(:another_model_with_created_at).to_a` is correct to be an error. I feel like that `.joins(:another_model_with_created_at)` is meaningless. My expectation (use case) is here:

https://gist.github.com/kamipo/5409fa4f54ebd2e4872d6b8edc2044ed

```ruby
relation = Job.order("id DESC").limit(20)
relation = relation.where(user: current_user, status: "success")
relation = relation.joins(:recent_job)
 
# this query succeed if excepted the order in the relation.
#
#   SELECT  1 AS one FROM `jobs` INNER JOIN `recent_jobs` ON `recent_jobs`.`job_id` = `jobs`.`id` WHERE `jobs`.`status` = 'success' AND `jobs`.`user_id` = 1 LIMIT 1
if relation.exists?
  # But originally this query is correct to be an error.
  #
  #   ActiveRecord::StatementInvalid: Mysql2::Error: Column 'id' in order clause is ambiguous: SELECT  `jobs`.* FROM `jobs` INNER JOIN `recent_jobs` ON `recent_jobs`.`job_id` = `jobs`.`id` WHERE `jobs`.`status` = 'success' AND `jobs`.`user_id` = 1  ORDER BY id DESC LIMIT 20
  relation.each { |record| process_record(record) } rescue nil
end
```

And I'm just curious about keeping the order is slow. This means that `relation.to_a` is slower due to keeping the order, so run `relation.exists?` at first to prevent slower `relation.to_a`, right? If so, I feel like that ideally it should add an index that optimize `relation.to_a`.